src/ucp/core/ucp_worker.c

```diff
@@ -1119,15 +1125,19 @@ void ucp_worker_iface_cleanup(ucp_worker_iface_t *wiface)
 
     ucp_worker_iface_disarm(wiface);
 
-    if (wiface->attr.cap.flags & UCP_WORKER_UCT_ALL_EVENT_CAP_FLAGS) {
+    if ((wiface->event_fd != -1) &&
```

Comment:
is there a case when event_fd is not -1, but uct_iface is NULL?

Reply:
yes, UCT iface is `NULL` when `ucp_worker_iface_open()` fails
because we do `ucp_worker_iface_init` (initializeds ) after the worker UCT ifaces were open

so, we can just check `wiface->iface == NULL` and if this is `true`, return from the function

Reply:
No, the question is opposite. Why do we need to check for `event_fd` if `uct_iface` is NULL?
We could call `ucp_worker_iface_cleanup` for non NULL ifaces only, like it was before. 
Why need to check uct_iface for NULL here, after the event_fd check?

Reply:
> Why need to check uct_iface for NULL here, after the event_fd check?

as discussed offline - fixed